docs/DeveloperGuide.md

```diff
@@ -720,40 +720,212 @@ testers are expected to do more *exploratory* testing.
+       also is tagged to the event in event index x)<br>
+       Expected: The persons with index y and z in the current person list are untagged from the event with index x in the current event list. A success message is shown in the
+       result display, indicating that the names of the persons with index y and z in the current person list has been untagged from the event title of the event with index x
+       in the current event list.
+
+    4. Test case: `untagEvent 0 p/1 2`<br>
+       Expected: No person is untagged from an event. Error message is shown in the result display.
+
+    5. Other incorrect tagEvent commands to try: `untagEvent x p/y z` (where either x is larger than the size of the current event list or y or z is
+       larger than the size of the current person list), `UNTAGEVENT 1 p/1 2`, `untagEvent 1 p/x y` (where either x or y is the index of a person that is not tagged in the first event
+       in the current event list).<br>
+       Expected: Similar to second test case.
+
+### Generating a pie chart statistic for an Event
+1. Generating a pie chart statistic from an Event
+    1. Prerequisites: An event in the current event list should have at least 1 tagged person.
+
+    2. Test case: `makeStats x t/g` (where x is the index of an event that has at least 1 tagged person)<br>
+       Expected: A new window is opened, showing a pie chart of the gender statistics for the event in event index x of the current event list.
+       Success message is shown in the result display.
+
+    3. Test case: `makeStats 0 t/g`<br>
+       Expected: No new window is opened. Error message is shown in the result display.
+
+    4. Other incorrect makeStats commands to try: `MAKESTATS 1 t/g`, `makeStats 1 t/g a`<br>
+       Expected: Similar to second test case.
+
+### Creating a mailing list for an Event
+1. Generating a mailing list from an Event
+    1. Prerequisites: An event in the current event list should have at least 1 tagged person.
+
+    2. Test case: `mailEvent 1`<br>
+       Expected: A CSV file with the name of the first event title of the first event is created in the /data folder.
+       Success message is shown in the result display.
+
+    3. Test case: `mailEventStats 0`<br>
+       Expected: No CSV files are created. Error message is shown in the result display.
+
+    4. Other incorrect makeStats commands to try: `MAILEVENT 1`, `mailEvent x` (where x is larger than the size of the displayed event list)<br>
+       Expected: Similar to second test case.
 
 ### Saving data
 
 1. Dealing with missing/corrupted data files
 
-   1. _{explain how to simulate a missing/corrupted file, and the expected behavior}_
+    1. Test case: Delete the data file manually when not inside the application (by deleting the addressbook.json file)<br>
+       Expected: Duke The Market launches with a set of sample data. Executing any valid command that updates data will create a new data file
```

Comment:
I though a new data file is created with sample data, and any valid command on the sample data just updates the data file? Or maybe I understand it wrongly

Reply:
I think its the method that I explained as I checked the data file when the new sample data is added however, the addressbook.json file is not created until I run a command